samples/snippets/csharp/VS_Snippets_CLR/verifyingahash/cs/program.cs

```diff
@@ -1,54 +1,32 @@
 ﻿//<Snippet1>
 using System;
+using System.Linq;
 using System.Security.Cryptography;
 using System.Text;
 
-class Class1
-{
-    static void Main()
-    {
-        //This hash value is produced from "This is the original message!"
-        //using SHA256.
-        byte[] sentHashValue = { 185, 203, 236, 22, 3, 228, 27, 130, 87, 23, 244, 15, 87, 88, 14, 43, 37, 61, 106, 224, 81, 172, 224, 211, 104, 85, 194, 197, 194, 25, 120, 217 };
-
-        //This is the string that corresponds to the previous hash value.
-        string messageString = "This is the original message!";
-
-        byte[] compareHashValue;
-
-        //Create a new instance of the UnicodeEncoding class to
-        //convert the string into an array of Unicode bytes.
-        UnicodeEncoding ue = new UnicodeEncoding();
+//This hash value is produced from "This is the original message!"
+//using SHA256.
+byte[] sentHashValue = Convert.FromHexString("67A1790DCA55B8803AD024EE28F616A284DF5DD7B8BA5F68B4B252A5E925AF79");
 
-        //Convert the string into an array of bytes.
-        byte[] messageBytes = ue.GetBytes(messageString);
+//This is the string that corresponds to the previous hash value.
+string messageString = "This is the original message!";
 
-        //Create a new instance of the SHA256 class to create
-        //the hash value.
-        SHA256 shHash = SHA256.Create();
+//Convert the string into an array of bytes.
+byte[] messageBytes = Encoding.UTF8.GetBytes(messageString);
 
-        //Create the hash value from the array of bytes.
-        compareHashValue = shHash.ComputeHash(messageBytes);
+//Create the hash value from the array of bytes.
+byte[] compareHashValue = SHA256.HashData(messageBytes);
 
-        bool same = true;
+//Compare the values of the two byte arrays.
+bool same = sentHashValue.SequenceEqual(compareHashValue);
```

Comment:
Does this compile w/o using spans?
```suggestion
bool same = sentHashValue.AsSpan().SequenceEqual(compareHashValue);
```

Reply:
Yes, it compiles; it's LINQ `SequenceEqual`. Using `AsSpan()` would be slightly more efficient (it avoids some type checks), but I think simpler code is more important here.

Reply:
Ah, I forgot the using for LINQ mentally.

> simpler code is more important here.

👍🏻